parsley-debug/jvm/src/main/scala/parsley/debugger/util/XCollectorJava.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 Parsley Contributors <https://github.com/j-mie6/Parsley/graphs/contributors>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+package parsley.debugger.util
+
+import scala.collection.mutable
+import scala.util.{Success, Try}
+
+import parsley.debugger.internal.Rename.MapAddAll
+import parsley.token.Lexer
+
+import parsley.internal.deepembedding.frontend.LazyParsley
+
+// Java-reflection powered XCollector. For backup purposes. Not meant to be used directly, but inherited from.
+// The setAccessible methods on private members is a deprecated action, so proceed with caution.
+// $COVERAGE-OFF$
+private [parsley] abstract class XCollectorJava extends CollectorImpl {
+    override val supported: Boolean = true
+
+    private def getParsers(obj: Any): Try[Map[LazyParsley[_], String]] = {
+        val accumulator: mutable.HashMap[LazyParsley[_], String] = new mutable.HashMap()
+
+        Try {
+            // FIXME: Type matching by name is very brittle...
+            val parserGetters = obj.getClass.getDeclaredMethods.filter { mth =>
+                Seq(Class.forName("parsley.Parsley"), Class.forName("parsley.internal.deepembedding.frontend.LazyParsley"))
+                    .contains(mth.getReturnType) && mth.getParameterCount == 0
+            }
+
+            parserGetters.foreach(_.setAccessible(true)) // FIXME: This is deprecated API and may be removed in a later Java version!
```

Comment:
Perhaps we just emit a warning to the user saying that the parser could not have its name extracted and to make it public, and avoid setting accessible?

Reply:
Is there a preferred method for emitting this warning other than outputting to `stderr`?

Reply:
That sounds fine. It's the debugger after all, so some extraneous output seems fine. In principle, I suppose you could also store the fact that the name couldn't be resolved because of access modifiers in the generated data: that way a front-end could render it with a ⚠️ or something?

Reply:
Ah. I wonder if we could work something else out to mitigate that? Because as I understand it, they will lack names outside of the JVM anyway. Would perhaps manually attaching names to those parsers internally work?

Reply:
No more than the prettyName already introduces?

Reply:
I suppose what could be done would be to provide a `private [parsley]` value of type `Map[LazyParsley[_], String]` into the lexer that manually defines all the names, if the manual updating (should their internal field names ever change) is alright.

Reply:
The issue is that there is a lot of them to do manually, and unlike with the visitor added, there isn't really a way to enforce that "all parser additions to `Lexer` must have their internal field names written here as well".

Reply:
I suppose I could for now remove the ability to get names from internal `Lexer` objects within `unsafeLexerObjects`, and also remove private accesses in `collectNames` (via `getParsers`; adding a warning when something is private).